### PR TITLE
Time history for ABL forcing

### PR DESCRIPTION
**Pull request by locascio-m**
Added input table for ABL forcing time, wind speed, and wind direction

### CODE REVIEW

**Comment by tonyinme** [comment]
hey @locascio-m , the PR is looking good. Can you please add the documentation as well?

**Comment by locascio-m** [comment]
My mistake, just added it

**Comment by locascio-m** [comment]
I adjusted the code in ABLForcing.cpp based on the warning from Lint-cppcheck. The other warnings there and the error in the Docker check are not related to code that I've modified though.

File: amr-wind/equation_systems/icns/source_terms/ABLForcing.cpp [hunk 1]
```diff
@@ -19,7 +20,26 @@ ABLForcing::ABLForcing(const CFDSim& sim) : m_time(sim.time())
     // TODO: Allow forcing at multiple heights
     pp_abl.get("abl_forcing_height", m_forcing_height);
     amrex::ParmParse pp_incflo("incflo");
-    pp_incflo.getarr("velocity", m_target_vel);
+
+    pp_abl.query("velocity_timetable", m_vel_timetable);
+    if (!m_vel_timetable.empty()) {
+        std::ifstream ifh(m_vel_timetable, std::ios::in);
```
Comment (gdeskos) [inline amr-wind/equation_systems/icns/source_terms/ABLForcing.cpp:26]: Make sure you include something to do the end of line. See here https://github.com/Exawind/amr-wind/blob/3d2e837f51a8c3a403c359d40264b21bf5c88850/amr-wind/utilities/tagging/CartBoxRefinement.cpp#L38

File: amr-wind/wind_energy/ABLFieldInit.cpp [hunk 2]
```diff
@@ -63,9 +77,20 @@ void ABLFieldInit::operator()(
...
     const amrex::Real rho_init = m_rho;
-    const amrex::Real umean = m_vel[0];
-    const amrex::Real vmean = m_vel[1];
-    const amrex::Real wmean = m_vel[2];
+
+    amrex::Real umean;
+    amrex::Real vmean;
+    amrex::Real wmean;
+    if (!m_vel_timetable.empty()) {
+        umean = m_vel_speed * std::cos(m_vel_dir);
```
Comment (gdeskos) [inline amr-wind/wind_energy/ABLFieldInit.cpp:85]: turn these back to const amrex::Real

**Comment by locascio-m** [comment]
Added a new commit to address @gdeskos comments

File: docs/sphinx/user/inputs_Momentum_Sources.rst [hunk 3]
```diff
@@ -67,3 +67,10 @@ Section: Momentum Sources
    
    Height in meters at which the flow is forced to maintain the freestream
    inflow velocities specified through :input_param:`incflo.velocity`.
+
+.. input_param:: ABLForcing.velocity_timetable
+
+   **type:** String, optional
+   
+   Input file name for table that lists time, wind speed, and wind direction
```
Comment (tonyinme) [inline docs/sphinx/user/inputs_Momentum_Sources.rst:75]: Can you specify the format of the table and the units of each variable?

**Review by tonyinme — COMMENTED** [review]
This looks good. Please address the suggestion in the documentation.

**Review by tonyinme — APPROVED** [review]

**Review by gdeskos — APPROVED** [review]

**Pull request merged**